core/src/mindustry/entities/abilities/ShieldArcAbility.java

```diff
@@ -19,25 +21,86 @@ public class ShieldArcAbility extends Ability{
     private static Vec2 paramPos = new Vec2();
     private static final Cons<Bullet> shieldConsumer = b -> {
         if(b.team != paramUnit.team && b.type.absorbable && paramField.data > 0 &&
-            !(b.within(paramPos, paramField.radius - paramField.width/2f) && paramPos.within(b.x - b.deltaX, b.y - b.deltaY, paramField.radius - paramField.width/2f)) &&
-            (Tmp.v1.set(b).add(b.deltaX, b.deltaY).within(paramPos, paramField.radius + paramField.width/2f) || b.within(paramPos, paramField.radius + paramField.width/2f)) &&
+            !(b.within(paramPos, paramField.radius - paramField.width) && paramPos.within(b.x - b.deltaX, b.y - b.deltaY, paramField.radius - paramField.width)) &&
+            (Tmp.v1.set(b).add(b.deltaX, b.deltaY).within(paramPos, paramField.radius + paramField.width) || b.within(paramPos, paramField.radius + paramField.width)) &&
             (Angles.within(paramPos.angleTo(b), paramUnit.rotation + paramField.angleOffset, paramField.angle / 2f) || Angles.within(paramPos.angleTo(b.x + b.deltaX, b.y + b.deltaY), paramUnit.rotation + paramField.angleOffset, paramField.angle / 2f))){
 
-            b.absorb();
-            Fx.absorb.at(b);
+            if(paramField.chanceDeflect > 0f && b.vel.len() >= 0.1f && b.type.reflectable && Mathf.chance(paramField.chanceDeflect)){
 
-            //break shield
+                //make sound
+                paramField.deflectSound.at(paramPos, Mathf.random(0.9f, 1.1f));
+
+                //translate bullet back to where it was upon collision
+                b.trns(-b.vel.x, -b.vel.y);
+
+                float penX = Math.abs(paramPos.x - b.x), penY = Math.abs(paramPos.y - b.y);
+
+                if(penX > penY){
+                    b.vel.x *= -1;
+                }else{
+                    b.vel.y *= -1;
+                }
+
+                b.owner = paramUnit;
+                b.team = paramUnit.team;
+                b.time += 1f;
+
+            }else{
+                b.absorb();
+                Fx.absorb.at(b);
+            }
+            
+            // break shield
             if(paramField.data <= b.damage()){
                 paramField.data -= paramField.cooldown * paramField.regen;
 
                 Fx.arcShieldBreak.at(paramPos.x, paramPos.y, 0, paramField.color == null ? paramUnit.type.shieldColor(paramUnit) : paramField.color, paramUnit);
             }
 
-            paramField.data -= b.damage();
+            // shieldDamage for consistency
+            paramField.data -= b.type.shieldDamage(b);
             paramField.alpha = 1f;
         }
     };
 
+    protected static final Cons<Unit> unitConsumer = unit -> {
+        // ignore core units
+        if(paramField.data > 0 && !unit.spawnedByCore &&
```

Comment:
It might be better to check if the unit is `targetable()` instead. Checking for core-spawned units will (mostly) work on Erekir, but it's unusual otherwise.